src/components/profile/profile-logged-in/profile-details.tsx

```
@@ -1,47 +1,36 @@
-import React, { useState } from "react";
+import React from "react";
 import { useI18nStore } from "../../../i18n/i18n-store";
-import InputFieldIcon from "../../input/input-field-icon";
 import TertiaryButton from "../../buttons/tertiary";
-import EditIcon from "../../icons/edit-icon";
+import { useAuthStore } from "../../../auth/auth-store";
+import EmailInputWithIcon from "../validation/email-input-with-icon";
+import UsernameInputWithIcon from "../validation/username-input-with-icon";
 
 const ProfileDetails: React.FC = () => {
   const i18n = useI18nStore().i18n();
-  const [usernameIsDisabled, setUsernameDisabled] = useState(true);
-  const [emailIsDisabled, setEmailDisabled] = useState(true);
+
+  const { forgotPassword, getUserData } = useAuthStore();
 
   return (
     <div className="md:shadow-gdk-soft mb-3 md:rounded-2xl md:border-2 md:p-7">
       <h2 className="text-2xl font-semibold">
         {i18n.navbar.profile.settings.subtitle}
       </h2>
 
-      <InputFieldIcon
+      <UsernameInputWithIcon
         label={i18n.navbar.profile.settings.username}
-        placeholder={i18n.navbar.profile.settings.placeholderUser}
-        onClick={() => {
-          setUsernameDisabled(!usernameIsDisabled);
-        }}
-        disabled={usernameIsDisabled}
-        setDisabled={setUsernameDisabled}
-        icon={<EditIcon />}
-      ></InputFieldIcon>
+        placeholder={getUserData()?.user_metadata.signup_username}
```

Your hooks throw errors but the errors are not handled here.